libraries/botbuilder-dialogs-adaptive-teams/src/actions/baseTeamsCacheInfoResponseDialog.ts

```diff
@@ -0,0 +1,125 @@
+/**
+ * @module botbuilder-dialogs-adaptive-teams
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+    BoolExpression,
+    BoolExpressionConverter,
+    Expression,
+    IntExpression,
+    IntExpressionConverter,
+    StringExpression,
+    StringExpressionConverter,
+} from 'adaptive-expressions';
+import { Converter, ConverterFactory, Dialog, DialogContext } from 'botbuilder-dialogs';
+import {
+    Activity,
+    ActivityTypes,
+    CacheInfo,
+    MessagingExtensionActionResponse,
+    MessagingExtensionResult,
+    StatusCodes,
+} from 'botframework-schema';
+import { MessagingExtensionResponse } from 'botbuilder';
+
+export interface BaseTeamsCacheInfoResponseDialogConfiguration {
+    disabled?: boolean | string | BoolExpression;
```

Comment:
This is perhaps a larger suggestion that could be resolved in a separate PR, but I'll bring it. I've seen these sorts of type unions all over the place, and it strikes me as a lot of unnecessary duplication. It seems like we have a set of "input types" that all can loosely be considered/converted to the same type. Maybe these should all be defined in a single place, such as:

```typescript
type BoolExpressionUnion = boolean | string | BoolExpression;
type StringExpressionUnion = string | Expression | StringExpression;
...
```

It might serve to simplify a lot of these interface definitions.

Reply:
I was unable to assign myself, but created an issue for it [here](https://github.com/microsoft/botbuilder-js/issues/3193), since this occurs in a LOT of the Adaptive libraries.